src/Features/LanguageServer/Protocol/Handler/Completion/CompletionHandler.cs

```diff
@@ -168,6 +168,13 @@ bool IsValidTriggerCharacterForDocument(Document document, char triggerCharacter
                 return completionItem;
             }
 
+            static string? GetInsertText(CompletionItem item, string label)
+            {
+                return (item.Properties.TryGetValue("InsertionText", out var insertText) && !insertText.Equals(label, StringComparison.Ordinal))
```

Comment:
this won't work.  as mentioned to @allisonchou there is no guarantee or reason that providers would set this particular code.  The most reliable way to get this info is to call GetChangeAsync and find the text that would be inserted.

Reply:
Which part of this won't work? This is the original implementation moved and with the added label equality check.

Reply:
this issue is present in the existing code, and would be resolved by https://github.com/dotnet/roslyn/pull/51166 once we fully enable the feature flag.  I wouldn't want to block this PR on it.